test/testHelpers.hpp

```diff
@@ -1417,4 +1417,40 @@ ::testing::AssertionResult assertArrayEq(std::string aName, std::string bName,
     return testWriteToOutputArray(aName, bName, a, b, metadata);
 }
 
+class TestData {
```

Comment:
Can you put this class in another file. I think this class is very specific to the approx functions because it is only working on floats. I can see the benefit of having a class like this because our current approach to reading test data is tedious. Perhaps we can come up with a better design later.